### DISCUSSION TITLE
Is the install prompt supposed to happen automatically?

### DISCUSSION THREAD

**Discussion by jsparks19** [khmyznikov/pwa-install#81] [Q&A] [answered]
If I do only what is in the Use section I'm not seeing any prompts to install the PWA. If I add code from the demo app with the show and hide buttons I can get it to pop up, though the install button doesn't seem to work. I kind of assumed this would handle all of that for me, but wondering if it's just providing the functions and we're responsible for triggering them as needed? It looks great, just need more clarity on what I need to implement to use it. Thanks!

**Comment by AndrasE** [comment]
same here, it looks amazing but i wish it would install on click bcus its just a modal at this point 😢

**Comment by khmyznikov** [comment, marked as answer]
@jsparks19 @AndrasE You can't call browser installation any time you want. Browser have heuristics to throttle this kind of requests until user makes at least some interactions on your site. Yes, install prompt happen automatically if you added the library in a right way. Make sure it's loads as soon as possible, otherwise you need to test async approach (described in readme)

**Reply by AndrasE** [reply]
thank you @khmyznikov for me the issue was that the BeforeInstallPromptEvent wasn`t captured but it was due to my apps setup  as there was a little timeout before the app runs hence by the time the package try to access this it was already undefined. Great looking package!

**Reply by khmyznikov** [reply]
Yep that's exactly why the async mode was made. Check the readme.
